server/__tests__/index_test.js

```diff
@@ -0,0 +1,27 @@
+const request = require('supertest');
+const app = require('../index')
+
+describe('GET /api/products', () => {
+    test('responds with json and 200 status', function (done) {
+        request(app)
+            .get('/api/products/')
+            .expect('Content-Type', /json/)
+            .expect(200, done);
+    });
+})
+describe('PUT /api/products', () => {
+    test('responds with json and 200 status', function (done) {
+        request(app)
+            .put('/api/products/1')
+            .send({
+                id: 1,
+                shop_quantity: 11,
+            })
+            .set('Accept', 'application/json')
+            .expect(200)
+            .end(function (err, res) {
```

Comment:
I'm not sure what the `.end` function is doing - never used `supertest` before. Why is it needed?

Reply:
Good question. I was following an example when writing this test. However, looking at the documentation, it says when using end():

> expect() assertions that fail will not throw - they will return the assertion as an error to the .end() callback.

Honestly, I'm not sure I understand. However, I removed it and added done() at the end and it still works